### PR TITLE
Do not rely on task variables to be available in the same task

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: Tasks/SonarQubePostTest/SonarQubeMetrics.ps1 [hunk 1]
```diff
@@ -1,3 +1,6 @@
+$script:qualityGateStatus = ""
```
Comment (bgavrilMS) [inline Tasks/SonarQubePostTest/SonarQubeMetrics.ps1]: I'm replacing task variables with script scoped variables. These are persisted for the duration of this script. Task variables use process env variables underneath, however the new core clr agent uses a separate process to run each task so setting and getting a task variable withing the same task no longer works.

File: Tests/L0/SonarQubePostTest/SonarQubeMetrics.ps1 [hunk 2]
```diff
@@ -6,26 +6,13 @@ param()
 . $PSScriptRoot\..\..\..\Tasks\SonarQubePostTest\SonarQubeMetrics.ps1
 . $PSScriptRoot\..\..\lib\Initialize-Test.ps1
 
-$distributedTaskContext = "context"
+#$distributedTaskContext = "context"
```
Comment (bgavrilMS) [inline Tests/L0/SonarQubePostTest/SonarQubeMetrics.ps1]: Will remove commented out code.

File: Tests/L0/SonarQubePostTest/SonarQubeMetrics.ps1 [hunk 3]
```diff
@@ -6,26 +6,13 @@ param()
...
 
-# Test 1 - this test ensures the quality gate status is read from the build variable MSBuild.SonarQube.QualityGateStatus if it exists
```
Comment (bgavrilMS) [inline Tests/L0/SonarQubePostTest/SonarQubeMetrics.ps1]: This test is no longer needed as I check caching in the other test

File: Tests/L0/SonarQubePostTest/SonarQubeMetrics.ps1 [hunk 4]
```diff
@@ -6,26 +6,13 @@ param()
...
-# Act
-$actualStatus = GetOrFetchQualityGateStatus
-
-# Assert
-Assert-AreEqual "OK" $actualStatus "The cached quality gate status should be returned" 
-
-# Cleanup
-Unregister-Mock Get-TaskVariable
 
 # Test 2 - this goes through the whole E2E - waiting for the SQ analysis to complete and querying for the quality gate status
```
Comment (bgavrilMS) [inline Tests/L0/SonarQubePostTest/SonarQubeMetrics.ps1]: not Test 2

File: Tests/L0/SonarQubePostTest/SonarQubeMetrics.ps1 [hunk 5]
```diff
@@ -6,26 +6,13 @@ param()
...
 
 # Arrange 
-Register-Mock Get-TaskVariable { $null } -- -Context $distributedTaskContext -Name 'MSBuild.SonarQube.QualityGateStatus'
+#Register-Mock Get-TaskVariable { $null } -- -Context $distributedTaskContext -Name 'MSBuild.SonarQube.QualityGateStatus'
```
Comment (eyalantzMS) [inline Tests/L0/SonarQubePostTest/SonarQubeMetrics.ps1]: Delete?

**Comment by eyalantzMS** [comment]
LGTM

**Comment by duncanpMS** [comment]
LGTM

**Pull request merged**